numba/cuda/reduction.py

```diff
@@ -193,7 +239,8 @@ def copy_strides(arr, n, stride, tpb):
         while size >= 16:
             # Find the closest size that is power of two
             p2size = 2 ** int(math.log(size, 2))
-            # Plan for p2size
+            # Generate a plan for invoking kernels to reduce the first p2size
+            # elements
             plan = self._plan(p2size, nbtype, init)
```

Comment:
The only reason to pre-populate the plans (https://github.com/gmarkall/numba/blob/reduce_comments/numba/cuda/reduction.py#L333-L338) is to avoid the compilation, which triggers a synchronization, during the reduction.